launch_ros/launch_ros/descriptions/ros_executable.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2021 Southwest Research Institute, All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# DISTRIBUTION A. Approved for public release; distribution unlimited.
+# OPSEC #4584.
+#
+# Delivered to the U.S. Government with Unlimited Rights, as defined in DFARS
+# Part 252.227-7013 or 7014 (Feb 2014).
+#
+# This notice must appear in all copies of this file and its derivatives.
+
+"""Module for a description of a ROS Executable."""
+
+from typing import Iterable
+from typing import Optional
+
+from launch import Action
+from launch import LaunchContext
+from launch import SomeSubstitutionsType
+from launch.descriptions import Executable
+
+from launch_ros.substitutions import ExecutableInPackage
+
+from ..descriptions import Node
+
+
+class RosExecutable(Executable):
+    """Describes an executable with ROS features which may be run by the launch system."""
+
+    def __init__(
+            self, *,
+            executable: Optional[SomeSubstitutionsType] = None,
+            package: Optional[SomeSubstitutionsType] = None,
+            nodes: Iterable[Node] = None,
+            ros_arguments: Optional[Iterable[SomeSubstitutionsType]] = None,
+            arguments: Optional[Iterable[SomeSubstitutionsType]] = None,
+            **kwargs
```

Comment:
@mlanting meta: I wonder if it would make sense to allow for "global" parameters and remapping rules here (i.e. without a node name prefix to scope them).

Reply:
@mlanting even more so, there was (and probably still is) some confusion around parameter file formats and how do each apply to `launch_ros` actions, mainly owing to the fact that the file format expected by `rcl` accounts for multiple nodes in the same process whereas the file format expected by `rclcpp_components` is bound to a single node.

We could potentially sort that out. `RosExecutable` would accept the former while `Node` descriptions would accept the latter.